flink-clients/src/main/java/org/apache/flink/client/FlinkPipelineTranslationUtil.java

```diff
@@ -40,8 +43,16 @@ public static JobGraph getJobGraph(
         FlinkPipelineTranslator pipelineTranslator =
                 getPipelineTranslator(userClassloader, pipeline);
 
-        return pipelineTranslator.translateToJobGraph(
-                pipeline, optimizerConfiguration, defaultParallelism);
+        JobGraph jobGraph =
+                pipelineTranslator.translateToJobGraph(
+                        pipeline, optimizerConfiguration, defaultParallelism);
+
+        Map<String, String> parallelismOverrides =
+                optimizerConfiguration.get(PipelineOptions.PARALLELISM_OVERRIDES);
+        jobGraph.getJobConfiguration()
+                .set(PipelineOptions.PARALLELISM_OVERRIDES, parallelismOverrides);
+
```

Comment:
I suggest use `getOptional` instead of `get` for fetching `PARALLELISM_OVERRIDES` to ensure that the 'jobConfiguration' only contains which is explicitly set by users like below.

`optimizerConfiguration.getOptional(PipelineOptions.PARALLELISM_OVERRIDES).ifPresent(map -> jobGraph.getJobConfiguration().set(PipelineOptions.PARALLELISM_OVERRIDES, map));`


Although the current jobConfiguration does not yet include all job-level configurations, in the long term, this jobConfiguration field should contain and only contain the job-level configuration items explicitly configured by the user.

Reply:
Thanks for the suggestion @JunRuiLee. I agree it better distinguishes between an unset configuration and a configured but empty map. I have updated the PR according to this.